### DISCUSSION TITLE
Stuff like Accolades

### DISCUSSION THREAD

**Discussion by Teifion** [beyond-all-reason/tachyon#12] [Ideas]
How would you like to handle things like Accolades? I'm talking about things where the server wants to send a client a prompt, do we want to limit it strictly to protocol stuff (so accolades would be part of Tachyon and thus adding or changing them becomes harder) or should there be something more generic?

@Jazcash

**Comment by Jazcash** [comment]
I think that will relate to whether we want Tachyon/Teiserver (and related infra) to be BAR-specific or more generic. For me personally, I only care about BAR. The new lobby is specifically tailored for BAR, and my goals and strategy are all the simpler for it. If you're happy to do the same, then I'd say that all the accolades can be defined in the protocol. If there ever comes a time where want to make it more flexible, e.g. for modders to add their own accolades, then we can worry about that then, but I think it's a better use of time to just hardcode them for now.

In regards to accolades, does that cover both "social" type accolades, like as in DOTA2 where you can commend allies for teamwork, friendliness etc, and also game awards, e.g. top damage, top econ etc?

**Reply by Teifion** [reply]
I'd like Tachyon as a whole to be somewhat flexible, at the very least useful to other Recoil based games.

Accolades will be social based things but it's just an example, what if we want to add in something requesting the player give a thumbs up/down if they enjoyed the game or not?

**Reply by Jazcash** [reply]
If you're asking about that specific accolade then it sounds good to me. What other kind of things were you thinking might be accolades, because I can only think of common things like commending a player for friendliness or leadership. If there's not too many, I think it's best to define all accolades in the protocol for now. If we get to a point where we're wanting to change accolades a lot then we can do the work to make them defined by the server, but again I don't think it's worth it unless we get to that point.

**Reply by Teifion** [reply]
Anything where an option should be presented to the user and to get a response from them. Accolades are the most obvious example but if we wanted to solicit generic feedback that'd go there too.

It might be we have an engine test host and subsequent to games on it we want to present players with a quick form about bugs they might have encountered. Primarily I'm looking to allow flexibility on the matter.

**Reply by Jazcash** [reply]
Ah ok. That sounds less like what accolades would refer to though, so maybe call it something like feedback. That kind of thing does make sense to be defined in the server, and the Lobby Client Application™ can be the thing that decides when it's an appropriate time to display it to the user.

**Reply by Teifion** [reply]
100%. I'm purely using Accolades as an example of:

- Server sends application message saying "present this input option"
- Application decides how to render it based on the parameters
- User inputs data
- Application sends it to server
- Server does something with it
